src/main/java/org/datadog/jenkins/plugins/datadog/logs/DatadogOutputStream.java

```diff
@@ -27,29 +27,77 @@ of this software and associated documentation files (the "Software"), to deal
 
 import hudson.console.ConsoleNote;
 import hudson.console.LineTransformationOutputStream;
+import hudson.model.AbstractBuild;
+import hudson.model.Run;
 
 import java.io.IOException;
 import java.io.OutputStream;
+import java.util.Collection;
+import java.util.Map;
+import java.util.WeakHashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class DatadogOutputStream extends LineTransformationOutputStream {
     private OutputStream delegate;
     private DatadogWriter writer;
+    private Run<?, ?> run;
+    private Pattern p;
 
+    private final static Map<AbstractBuild<?, ?>, Collection<String>> secretsForBuild = new WeakHashMap<>();
 
     public DatadogOutputStream(OutputStream delegate, DatadogWriter writer) {
         super();
         this.delegate = delegate;
         this.writer = writer;
     }
 
+    public DatadogOutputStream(OutputStream delegate, DatadogWriter writer, Run<?, ?> run) {
+        super();
+        this.delegate = delegate;
+        this.writer = writer;
+        this.run = run;
+    }
+
+    public static Pattern getPatternForBuild(AbstractBuild<?, ?> build) {
+        if (secretsForBuild.containsKey(build)) {
+            return getAggregateSecretPattern(secretsForBuild.get(build));
+        } else {
+            return null;
+        }
+    }
+
+    public static Pattern getAggregateSecretPattern(Collection<String> inputs) {
+        // dummy input, would eventually need to scale to all forms of Secrets
+        String pattern = "hello";
```

Comment:
Currently just have pattern be the word hello but will find out how to get actual credentials